From ecashin@coraid.com Thu Jan 19 11:11:56 2006
Message-ID: <d856359f3883ba1b7f11683932a478be@coraid.com>
From: "Ed L. Cashin" <ecashin@coraid.com>
Date: Thu, 19 Jan 2006 13:46:25 -0500
CC: ecashin@coraid.com, Greg K-H <greg@kroah.com>
Subject: [PATCH 2.6.15-git9] aoe [5/8]: allow network interface migration on packet retransmit

Retransmit to the current network interface for an AoE device.

Signed-off-by: "Ed L. Cashin" <ecashin@coraid.com>

---
 drivers/block/aoe/aoecmd.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/block/aoe/aoecmd.c
+++ gregkh-2.6/drivers/block/aoe/aoecmd.c
@@ -286,6 +286,8 @@ rexmit(struct aoedev *d, struct frame *f
 	h = (struct aoe_hdr *) f->data;
 	f->tag = n;
 	h->tag = cpu_to_be32(n);
+	memcpy(h->dst, d->addr, sizeof h->dst);
+	memcpy(h->src, d->ifp->dev_addr, sizeof h->src);
 
 	skb = skb_prepare(d, f);
 	if (skb) {
